In [1]:
import pandas as pd

In [7]:
data = pd.read_csv("C:\\Users\\nilka\\OneDrive\\Desktop\\excelr\\self paced course\\data science assignment\\1.assignments\\3.hypothesis testing\\Customerorderform.csv")

In [8]:
data.head()

Phillippines   Indonesia       Malta       India
0   Error Free  Error Free   Defective  Error Free
1   Error Free  Error Free  Error Free   Defective
2   Error Free   Defective   Defective  Error Free
3   Error Free  Error Free  Error Free  Error Free
4   Error Free  Error Free   Defective  Error Free

# Problem Statement

To check whether defective % varies by centers

# Hypothesis statement


H0: There is no difference in defective percentage by centers.

H1: There is difference in defective percentage by centers.

In [21]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 300 entries, 0 to 299
Data columns (total 4 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   Phillippines  300 non-null    object
 1   Indonesia     300 non-null    object
 2   Malta         300 non-null    object
 3   India         300 non-null    object
dtypes: object(4)
memory usage: 9.5+ KB


In [22]:
data.Phillippines.value_counts(),data.Indonesia.value_counts(),data.Malta.value_counts(),data.India.value_counts()

(Error Free    271
 Defective      29
 Name: Phillippines, dtype: int64,
 Error Free    267
 Defective      33
 Name: Indonesia, dtype: int64,
 Error Free    269
 Defective      31
 Name: Malta, dtype: int64,
 Error Free    280
 Defective      20
 Name: India, dtype: int64)

In [23]:
from scipy import stats
stats.chi2_contingency([data.Phillippines.value_counts(),data.Indonesia.value_counts(),data.Malta.value_counts(),data.India.value_counts()])

(3.8589606858203545,
 0.2771020991233144,
 3,
 array([[271.75,  28.25],
        [271.75,  28.25],
        [271.75,  28.25],
        [271.75,  28.25]]))

P value =0.277

As P > 0.05 we accept H0   thus  there is no difference in defective percentage by centers